src/combinatorics/simple_graph/density.lean

```diff
@@ -122,6 +123,106 @@ by rw [edge_density, finset.card_empty, nat.cast_zero, zero_mul, div_zero]
 @[simp] lemma edge_density_empty_right (s : finset α) : edge_density r s ∅ = 0 :=
 by rw [edge_density, finset.card_empty, nat.cast_zero, mul_zero, div_zero]
 
+lemma card_interedges_finpartition_left [decidable_eq α] (P : finpartition s) (t : finset β) :
+  (interedges r s t).card = ∑ a in P.parts, (interedges r a t).card :=
+begin
+  classical,
+  simp_rw [←P.bUnion_parts, interedges_bUnion_left, id.def],
+  rw card_bUnion,
+  exact λ x hx y hy h, interedges_disjoint_left r (P.disjoint hx hy h) _,
+end
+
+lemma card_interedges_finpartition_right [decidable_eq β] (s : finset α) (P : finpartition t) :
+  (interedges r s t).card = ∑ b in P.parts, (interedges r s b).card :=
+begin
+  classical,
+  simp_rw [←P.bUnion_parts, interedges_bUnion_right, id],
+  rw card_bUnion,
+  exact λ x hx y hy h, interedges_disjoint_right r _ (P.disjoint hx hy h),
+end
+
+lemma card_interedges_finpartition [decidable_eq α] [decidable_eq β] (P : finpartition s)
+  (Q : finpartition t) :
+  (interedges r s t).card = ∑ ab in P.parts.product Q.parts, (interedges r ab.1 ab.2).card :=
```

Comment:
I was about to suggest writing `P.parts ×ˢ Q.parts` but then I noticed we do not have:
```
instance {α β : Type*} : has_set_prod (finset α) (finset β) (finset (α × β)) := ⟨finset.product⟩
```
It might be nice if someone added this :-)

Reply:
I once tried adding this, but the number of coercions you end up needing doesn't seem to be worth it. Maybe we could turn `has_set_prod` into notation?